src/encoding/src/rlp.cairo

```diff
@@ -0,0 +1,183 @@
+use alexandria_data_structures::array_ext::ArrayTraitExt;
+use alexandria_numeric::integers::U32Trait;
+
+// Possible RLP types
+#[derive(Drop, PartialEq)]
+enum RLPType {
+    String,
+    List
+}
+
+#[derive(Drop, Copy, PartialEq)]
+enum RLPItem {
+    String: Span<u8>,
+    List: Span<RLPItem>
+}
+
+#[generate_trait]
+impl RLPImpl of RLPTrait {
+    /// Returns RLPType from the leading byte with
+    /// its offset in the array as well as its size.
+    ///
+    /// # Arguments
+    /// * `input` - Array of byte to decode
+    /// # Returns
+    /// * `(RLPType, offset, size)` - A tuple containing the RLPType
+    /// the offset and the size of the RLPItem to decode
+    /// # Errors
+    /// * empty input - if the input is empty
+    /// * input too short - if the input is too short for a given
+    fn decode_type(input: Span<u8>) -> (RLPType, u32, u32) {
```

Comment:
I think this should return a result instead of panics

Reply:
Ho ok ! Will do !